fem/gridfunc.cpp

```diff
@@ -4162,6 +4200,368 @@ double ZZErrorEstimator(BilinearFormIntegrator &blfi,
    return std::sqrt(total_error);
 }
 
+void TensorProductLegendre(int dim,                // input
+                           int order,              // input
+                           const Vector &x_in,     // input
+                           const Vector &xmax,     // input
+                           const Vector &xmin,     // input
+                           Vector &poly,           // output
+                           double angle,           // input (optional)
+                           const Vector *midpoint) // input (optional)
+{
+   MFEM_VERIFY(dim >= 1, "dim must be positive");
+   MFEM_VERIFY(dim <= 3, "dim cannot be greater than 3");
+   MFEM_VERIFY(order >= 0, "order cannot be negative");
+
+   bool rotate = (angle != 0.0) || (midpoint->Norml2() != 0.0);
+
+   Vector x(dim);
+   if (rotate && dim == 2)
+   {
+      // Rotate coordinates to match rotated bounding box
+      Vector tmp(dim);
+      tmp = x_in;
+      tmp -= *midpoint;
+      x[0] = tmp[0]*cos(angle) - tmp[1]*sin(angle);
+      x[1] = tmp[0]*sin(angle) + tmp[1]*cos(angle);
+   }
+   else
+   {
+      // Bounding box is not reorientated no need to change orientation
+      x = x_in;
+   }
+
+   // Map x to [0, 1] to use CalcLegendre since it uses shifted Legendre Polynomials.
+   double x1 = (x(0) - xmin(0))/(xmax(0)-xmin(0)), x2, x3;
+   Vector poly_x(order+1), poly_y(order+1), poly_z(order+1);
+   Poly_1D poly1d;
```

Comment:
This `poly1d` variable shadows a global declared in `fem/fe/fe_base.hpp`. I believe you should be able to make use of the global here rather than declaring it locally. Can you double-check this and make the necessary changes?